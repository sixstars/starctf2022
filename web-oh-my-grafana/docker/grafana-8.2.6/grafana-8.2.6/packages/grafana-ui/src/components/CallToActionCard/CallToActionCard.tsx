import React from 'react';
import { css, cx } from '@emotion/css';
import { useStyles2 } from '../../themes/ThemeContext';
import { GrafanaTheme2 } from '@grafana/data';

export interface CallToActionCardProps {
  message?: string | JSX.Element;
  callToActionElement: JSX.Element;
  footer?: string | JSX.Element;
  className?: string;
}

export const CallToActionCard: React.FunctionComponent<CallToActionCardProps> = ({
  message,
  callToActionElement,
  footer,
  className,
}) => {
  const css = useStyles2(getStyles);

  return (
    <div className={cx([css.wrapper, className])}>
      {message && <div className={css.message}>{message}</div>}
      {callToActionElement}
      {footer && <div className={css.footer}>{footer}</div>}
    </div>
  );
};

const getStyles = (theme: GrafanaTheme2) => ({
  wrapper: css`
    label: call-to-action-card;
    padding: ${theme.spacing(3)};
    background: ${theme.colors.background.secondary};
    border-radius: ${theme.shape.borderRadius(2)};
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    flex-grow: 1;
  `,
  message: css`
    margin-bottom: ${theme.spacing(3)};
    font-style: italic;
  `,
  footer: css`
    margin-top: ${theme.spacing(3)}};
  `,
});
