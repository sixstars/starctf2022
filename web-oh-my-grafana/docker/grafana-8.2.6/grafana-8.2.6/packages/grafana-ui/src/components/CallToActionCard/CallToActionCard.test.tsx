import React from 'react';
import { render } from 'enzyme';
import { CallToActionCard } from './CallToActionCard';

describe('CallToActionCard', () => {
  describe('rendering', () => {
    it('when no message and footer provided', () => {
      const tree = render(<CallToActionCard callToActionElement={<a href="http://dummy.link">Click me</a>} />);
      expect(tree).toMatchSnapshot();
    });

    it('when message and no footer provided', () => {
      const tree = render(
        <CallToActionCard
          message="Click button bellow"
          callToActionElement={<a href="http://dummy.link">Click me</a>}
        />
      );
      expect(tree).toMatchSnapshot();
    });

    it('when message and footer provided', () => {
      const tree = render(
        <CallToActionCard
          message="Click button bellow"
          footer="footer content"
          callToActionElement={<a href="http://dummy.link">Click me</a>}
        />
      );
      expect(tree).toMatchSnapshot();
    });
  });
});
