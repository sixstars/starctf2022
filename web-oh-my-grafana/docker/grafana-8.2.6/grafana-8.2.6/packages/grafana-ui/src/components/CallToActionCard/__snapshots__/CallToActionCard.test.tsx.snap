// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallToActionCard rendering when message and footer provided 1`] = `
<div
  class="css-xtd6ws-call-to-action-card"
>
  <div
    class="css-m2iibx"
  >
    Click button bellow
  </div>
  <a
    href="http://dummy.link"
  >
    Click me
  </a>
  <div
    class="css-11lqfaj"
  >
    footer content
  </div>
</div>
`;

exports[`CallToActionCard rendering when message and no footer provided 1`] = `
<div
  class="css-xtd6ws-call-to-action-card"
>
  <div
    class="css-m2iibx"
  >
    Click button bellow
  </div>
  <a
    href="http://dummy.link"
  >
    Click me
  </a>
</div>
`;

exports[`CallToActionCard rendering when no message and footer provided 1`] = `
<div
  class="css-xtd6ws-call-to-action-card"
>
  <a
    href="http://dummy.link"
  >
    Click me
  </a>
</div>
`;
